src/components/NewDesign/Swap/Swap.tsx

```diff
@@ -0,0 +1,424 @@
+        printBNtoBN(amountTo, tokens[tokenToIndex].decimal).gt(
+          tokens[tokenToIndex].balance
+        )
+      ) {
+        return 'Insufficient balance'
+      }
+    } else {
+      if (
+        printBNtoBN(amountFrom, tokens[tokenFromIndex].decimal).gt(
+          tokens[tokenFromIndex].balance
+        )
+      ) {
+        return 'Insufficient balance'
+      }
+    }
+    return 'Swap'
+  }
+
+  const setSlippage = (slippage: string): void => {
+    setSlippTolerance(slippage)
+  }
+
+  const handleClickSettings = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+    blurContent()
+    setSettings(true)
+  }
+
+  const hoverDetails = () => {
+    setDetails(!details)
+  }
+
+  const handleCloseSettings = () => {
+    unblurContent()
+    setSettings(false)
+  }
+  return (
+    <Grid container>
+      <Grid container className={classes.header}>
+        <Typography component='h1'>Swap tokens</Typography>
+        <CardMedia image={settingIcon} className={classes.settingsIcon} onClick={handleClickSettings}/>
+        <Grid className={classes.slippage}>
+          <Slippage open={settings}
+            setSlippage={setSlippage}
+            handleClose={handleCloseSettings}
+            anchorEl={anchorEl}/>
+        </Grid>
+      </Grid>
+      <Grid container className={classes.root} direction='column'>
+        <Box className={classes.tokenComponentTextContainer}>
+          <Typography className={classes.tokenComponentText}>Est.: </Typography>
+          <Typography className={classes.tokenComponentText}>
+          Balance: {tokenFromIndex !== null
+              ? swap
+                ? printBN(tokens[tokenToIndex ?? 0].balance, tokens[tokenToIndex ?? 0].decimal)
+                : printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal) : '0'}
+          </Typography>
+        </Box>
+        <ExchangeAmountInput
+          value={amountFrom}
+          className={classes.amountInput}
+          style={{ transform: swap !== null ? swap ? 'translateY(104px)' : 'translateY(0px)' : '' }}
+          setValue={value => {
+            if (value.match(/^\d*\.?\d*$/)) {
+              setAmountFrom(value)
+              updateEstimatedAmount((+value * tax).toString())
+            }
+          }}
+          placeholder={'0.0'}
+          onMaxClick={() => {
+            if (tokenToIndex !== null && tokenFromIndex !== null) {
+              if (swap) {
+                setAmountTo(printBN(tokens[tokenToIndex].balance, tokens[tokenToIndex].decimal))
+                updateFromEstimatedAmount(printBN(tokens[tokenToIndex].balance, tokens[tokenToIndex].decimal))
+              } else {
+                setAmountFrom(printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal))
+                updateEstimatedAmount(printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal))
+              }
+            }
+          }}
+          tokens={swap ? tokenY.map((
```

Comment:
maps here are not necessary, you can use type with more props as value of variable of type with less props

Reply:
done